domain/storageprovisioning/service/filesystem.go

```diff
@@ -0,0 +1,187 @@
+
+	// InitialWatchStatementMachineProvisionedFilesystems returns both the
+	// namespace for watching filesystem life changes where the filesystem is
+	// machine provisioned. On top of this the initial query for getting all
+	// filesystems in the model that are machine provisioned is returned.
+	//
+	// Only filesystems that can be provisioned by the machine connected to the
+	// supplied net node will be emitted.
+	InitialWatchStatementMachineProvisionedFilesystems(netNodeUUID string) (string, eventsource.Query[map[string]life.Life])
+
+	// InitialWatchStatementModelProvisionedFilesystems returns both the
+	// namespace for watching filesystem life changes where the filesystem is
+	// model provisioned. On top of this the initial query for getting all
+	// filesystems in the model that are model provisioned is returned.
+	InitialWatchStatementModelProvisionedFilesystems() (string, eventsource.NamespaceQuery)
+
+	// InitialWatchStatementMachineProvisionedFilesystemAttachments returns
+	// both the namespace for watching filesystem attachment life changes where
+	// the filesystem attachment is machine provisioned. On top of this the
+	// initial query for getting all filesystem attachments in the model that
+	// are machine provisioned is returned.
+	//
+	// Only filesystem attachments that can be provisioned by the machine
+	// connected to the supplied net node will be emitted.
+	InitialWatchStatementMachineProvisionedFilesystemAttachments(netNodeUUID string) (string, eventsource.Query[map[string]life.Life])
+
+	// InitialWatchStatementModelProvisionedFilesystemAttachments returns both
+	// the namespace for watching filesystem attachment life changes where the
+	// filesystem attachment is model provisioned. On top of this the initial
+	// query for getting all filesystem attachments in the model that are model
+	// provisioned is returned.
+	InitialWatchStatementModelProvisionedFilesystemAttachments() (string, eventsource.NamespaceQuery)
+}
+
+// GetFilesystemAttachmentIDs returns the
+// [storageprovisioning.FilesystemAttachmentID] information for each of the
+// supplied filesystem attachment uuids. If a filesystem attachment does exist
+// for a supplied uuid or if a filesystem attachment is not attached to either a
+// machine or unit then this uuid will be left out of the final result.
+//
+// It is not considered an error if a filesystem attachment uuid no longer
+// exists as it is expected the caller has already satisfied this requirement
+// themselves.
+//
+// This function exists to help keep supporting storage provisioning facades
+// that have a very week data model about what a filesystem attachment is
+// attached to.
+//
+// All returned values will have either the machine name or unit name value
+// filled out in the [storageprovisioning.FilesystemAttachmentID] struct.
+func (s *Service) GetFilesystemAttachmentIDs(
+	ctx context.Context, uuids []string,
```

Comment:
Do we want to introduce some storage UUID types (for example, FilesystemAttachmentUUID) instead of using strings?

Reply:
no this is called from inside of a strings watcher and exists because of our facade's poor design. Doing this would add overhead and complexity to the problem.